python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+
+from torch import Tensor
+
+from kserve.logging import logger
+import pathlib
+from typing import Dict, Union, Tuple, Any
+
+from kserve.errors import InferenceError
+from kserve.storage import Storage
+
+from kserve.protocol.infer_type import InferRequest, InferResponse, InferInput
+from kserve.utils.utils import get_predict_input, get_predict_response
+from kserve import Model
+import torch
+from transformers import AutoModelForCausalLM, AutoModelForSeq2SeqLM, AutoTokenizer, \
+    AutoConfig, \
+    AutoModelForSequenceClassification, AutoModelForTokenClassification, AutoModelForQuestionAnswering, \
+    AutoModelForMaskedLM, BatchEncoding
+
+ARCHITECTURES_2_TASK = {
```

Comment:
Consider moving to enum or constant class

Reply:
created a task enum in task.py